src/Frame.js

```diff
@@ -0,0 +1,49 @@
+function Frame(finalFrame) {
+this._frameRoll1Score = 0;
+this._frameRoll2Score = 0;
+this._frameRoll3Score = 0;
+this._bonusScore = 0;
+this._strike = false;
+this._done = false;
+this._spare = false;
+this._finalFrame = finalFrame;
+this._rollNumber = 0;
+
+Frame.prototype.totalScoreForFrame = function(){
+  if (this._done) {
+    return this._frameRoll1Score + this._frameRoll2Score + this._frameRoll3Score + this._bonusScore;
+  }
+  return 0;
+};
+
+Frame.prototype.roll = function(pins_hit) {
```

Comment:
This method is quite long, with many if/else's. I'm sure you can simplify it further.

Reply:
One way to do that may be private methods (actually, JS does not have private methods, but by convention, an method name starting with `_` will be considered private).

For example, this method could be:
```
Frame.prototype.roll = function(pins_hit) {
    if (this._done) return;

    if (this._rollNumber === 0) {
    	this._addFirstRoll(pins_hit);
    } else if (this._rollNumber === 1) {
    	this._addSecondRoll(pins_hit);
    } else {
      this._frameRoll3Score = pins_hit;
      this._done = true;
    };
}
```

And then the rest could be separated into the private methods `_addFirstRoll` and `_addSecondRoll`